### ISSUE TITLE
ActivatedRoute.data observable emitting linearly increasing number of events to child route on route change.

### ISSUE THREAD

**Issue by rrfenton** [angular/angular#13215]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
When a user is subscribed to the ActivatedRoute.data property of a child route that has a variable as it's route identifier, "e.g. {path: ":childRouteId"}, the amount of events emitted on each route change to the ActivateRoute.data subscription rises at a linear rate, causing the subscription to fire more times as the user navigates around children of the parent route.

So, for example, lets say I have 3 routes, child1, child2, and child3.

If I navigate to child1, my ActivatedRoute.data subscription will fire 1 time on route change.
If I navigate to child2 from child1, my ActivatedRoute.data subscription will fire 3 times on route change.
If I navigate to child3 from child2, my ActivatedRoute.data subscription will fire 4 times on route change.

This happens even if I have no new resolved data.

The ActivatedRoute.params route fires only once on every route change.

**Expected behavior**
I would expect that ActivatedRoute.data would act exactly like params, or more accurately, only fire based on the resolvers for that route, as said by the comments in router_state.d.ts

     * The static and resolved data of this route. The observable will emit a new value when
     * any of the resolvers returns a new object.

**Minimal reproduction of the problem with instructions**
Here's a demo of the bug in question:
http://plnkr.co/edit/QVIqMWQ7gziqYPMlY0Tf?p=preview

Essentially have a parent route, and a child route.  Have the parent route trigger navigation to the children routes, and observe the amount of times the data subscription is fired.

**What is the motivation / use case for changing the behavior?**
I currently need to use data that comes back from the resolver to build an HTTP call.  I want to show a spinner while the data call is loading information, which means I need to make the call from the component, as I can't update component level properties from the resolve service.

Since the ActivatedRoute.data observable is firing multiple times, I am making the same http call multiple times needlessly.  Workarounds are available for this problem, but it doesn't solve the underlying issue of the number of times this observable fires.

**Please tell us about your environment:**

OS: Windows 10
IDE: Visual Studio Code
Package Manager: Webpack

* **Angular version:** 2.1.1

* **Browser:** [Chrome 54 ]

 
* **Language:** [TypeScript 2.0.x ]

**Comment by robwormald** [comment]
This is works as intended - it's not that the router is firing multiple events, its that you're wiring up a new subscription each time you return inside your outer flatMap, so each time you navigate, you're seeing multiple emits.

Depending on how you want to handle your data flow, I'd suggest either using `.take(1)` on the inner Observable, or combining the two, eg:

```ts
Observable.combineLatest(route.params, route.data, (params, data) => { ... })
```

**Comment by robwormald** [comment]
You can see this in action http://plnkr.co/edit/ba8v1GjWdHdzg2MNk5Xs?p=preview

**Comment by rrfenton** [comment]
Thanks for the help.

@robwormald  turns out you were on the nose regarding multiple emits from the second obserable because there were multiple instances of that subscriber.  That really caught me off guard since I was under the assumption that the mergemap would combine those observables, instead of creating a child instance.

I ended up using an  Observable.combineLatest() since I needed to know when either the params or data would update to create subsequent requests.

Thanks and sorry for taking up your time!
